runtime/compiler/env/J9KnownObjectTable.hpp

```diff
@@ -59,13 +59,24 @@ class OMR_EXTENSIBLE KnownObjectTable : public OMR::KnownObjectTableConnector
 
    KnownObjectTable(TR::Compilation *comp);
 
+   TR::KnownObjectTable *self();
+
    virtual Index getEndIndex();
    virtual Index getIndex(uintptrj_t objectPointer);
+   Index getIndex(uintptrj_t objectPointer, bool isArrayWithConstantElements);
    virtual uintptrj_t *getPointerLocation(Index index);
    virtual bool isNull(Index index);
 
    virtual void dumpTo(TR::FILE *file, TR::Compilation *comp);
 
+   virtual Index getIndexAt(uintptrj_t *objectReferenceLocation);
+   virtual Index getIndexAt(uintptrj_t *objectReferenceLocation, bool isArrayWithConstantElements);
+   virtual Index getExistingIndexAt(uintptrj_t *objectReferenceLocation);
+
+   virtual uintptrj_t getPointer(Index index);
```

Comment:
These existing methods have become virtual. Why? I thought that the extensible class framework was supposed to use static inheritance.

Reply:
Will clean up the `virtual` keywords. I guess originally, `self()` is not implemented in  `OpenJ9` and `virtual` still mattered.